### PR TITLE
Psss syndrome map overlays

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
[#2107](https://github.com/beyondessential/tupaia-backlog/issues/2107)
[#2108](https://github.com/beyondessential/tupaia-backlog/issues/2108)
### Changes:
- Support custom popup label (new config: `customLabel`)
- Migration for two map overlays

### CODE REVIEW

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    defaultTimePeriod: {
+      unit: 'day',
+      offset: 0,
+    },
+    periodGranularity: 'one_day_at_a_time',
+    customLabel,
+  },
+});
+
+const syndromeHeatMapOverlay = (syndromeName, dateElement, SyndromeCode, customLabel) => ({
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:87]: Very minor but just for naming consistency:
```suggestion
const syndromeHeatMapOverlay = (syndromeName, dateElement, syndromeCode, customLabel) => ({
```

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+const syndromeBaseMapOverlay = {
+  userGroup: 'PSSS Tupaia',
+  isDataRegional: true,
+  dataElementCode: null,
+  measureBuilder: 'sumLatestPerOrgUnit',
+  countryCodes: '{"PW"}',
+  projectCodes: '{"psss"}',
+};
+
+const syndromeRadiusMapOverlay = (SyndromeName, dateElement, SyndromeCode, customLabel) => ({
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:63]: Very minor but just for naming consistency:
```suggestion
const syndromeRadiusMapOverlay = (syndromeName, dateElement, syndromeCode, customLabel) => ({
```

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  ...syndromeBaseMapOverlay,
+  id: `PSSS_PW_${SyndromeCode}_Syndrome_Bubble_Radius`,
+  name: SyndromeName,
+  measureBuilderConfig: {
+    dataElementCodes: [dateElement],
+    entityAggregation: {
+      aggregationType: 'SUM_PER_ORG_GROUP',
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:70]: If you already use `sumLatestPerOrgUnit` measureBuilder (which will get the most recent data (only 1) per dataElement/orgUnit combo), do we also need this aggregation? (which I think will just sum data per dataElement/orgUnit combo again, which I think is redundant?)

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+});
+
+const syndromeHeatMapOverlay = (syndromeName, dateElement, SyndromeCode, customLabel) => ({
+  ...syndromeBaseMapOverlay,
+  id: `PSSS_PW_${SyndromeCode}_Syndrome_Heat_Map`,
+  name: syndromeName,
+  measureBuilderConfig: {
+    dataElementCodes: [dateElement],
+    entityAggregation: {
+      aggregationType: 'SUM_PER_ORG_GROUP',
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:94]: Same comment like above. Do we need `aggregationType` here?

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  }
+};
+
+exports.down = async function (db) {
+  await db.runSql(
+    `
+      DELETE FROM "map_overlay_group_relation" r
+      USING "map_overlay_group" g 
+      WHERE g.id = r.map_overlay_group_id
+      AND g.code in ('${radiusMapOverlayGroup.code}','${radiusMapOverlayGroup.code}');
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:165]: Minor but we're having 2 similar ids in WHERE IN here..?
```suggestion
      AND g.code in ('${radiusMapOverlayGroup.code}');
```

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+      DELETE FROM "map_overlay_group"
+      WHERE code in ('${radiusMapOverlayGroup.code}','${radiusMapOverlayGroup.code}');
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:168]: Minor but we're having 2 similar ids in WHERE IN here..?
```suggestion
      WHERE code in ('${radiusMapOverlayGroup.code}');
```

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+  );
+
+  for (const { name, dataElement, code, customLabel } of syndromes) {
+    const radiusMapOverlay = syndromeRadiusMapOverlay(name, dataElement, code ?? name, customLabel);
+    const heatMapOverlay = syndromeHeatMapOverlay(name, dataElement, code ?? name, customLabel);
+    for (const mapOverlay of [radiusMapOverlay, heatMapOverlay]) {
+      await db.runSql(
+        `
+          DELETE FROM "mapOverlay"
+          WHERE id in ('${mapOverlay.id}','${mapOverlay.id}');
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:179]: Minor but we're having 2 similar ids in WHERE IN here..?
```suggestion
          WHERE id in ('${mapOverlay.id}');
```

File: packages/web-config-server/src/apiV1/measureData.js [hunk 8]
```diff
@@ -243,6 +247,7 @@ export default class extends DataAggregatingRouteHandler {
       hideFromMenu,
       hideFromLegend,
       hideFromPopup,
+      customLabel,
```
Comment (chris-bes) [inline packages/web-config-server/src/apiV1/measureData.js:250]: Nice <3

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+    name: 'Conjunctivitis',
+    code: 'CON',
+    customLabel: 'Conjunctivitis Case Numbers',
+  },
+];
+
+const syndromeBaseMapOverlay = {
+  userGroup: 'PSSS Tupaia',
+  isDataRegional: true,
+  dataElementCode: null,
```
Comment (chris-bes) [inline packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:57]: As I remember for a measureBuilder, if you're specifying the dataElementCodes in the measureBuilderConfig, you have to set `dataElementCode` column to `value`, then the front end will know to pick the value from the measureData. (You can check other mapOverlays `select * from "mapOverlay" where "measureBuilder" = 'sumLatestPerOrgUnit'`)

```suggestion
  dataElementCode: 'value',
```

**Review by chris-bes — CHANGES_REQUESTED** [review]
Looks good! Just a few minor suggestions

File: packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js (hunk 9 above)
Comment (billli0) [reply packages/database/src/migrations/20210421001607-PsssPalauSyndromBubbleMapOverlay-modifies-data.js:57]: Thanks Chris, I was just testing if this value changes will effect the code, and forgot to change it back.. (result is that can be changed to anything)

**Comment by billli0** [comment]
Thanks Chris, sorry for letting you read lots of the duplicated variable things.

**Review by chris-bes — APPROVED** [review]
LGTM!

**Pull request merged**